src/plots/cartesian/layout_attributes.js

```diff
@@ -1003,6 +1003,28 @@ module.exports = {
             'Only has an effect if `anchor` is set to *free*.'
         ].join(' ')
     },
+    autoshift: {
+        valType: 'boolean',
+        dflt: false,
+        editType: 'plot',
+        description: [
+            'Automatically reposition the axis to avoid',
+            'overlap with other axes with the same `overlaying` value.',
+            'This repositioning will account for any `shift` amount applied to other',
+            'axes on the same side with `autoshift=true`.',
+            'Only has an effect if `anchor` is set to *free*.',
+        ].join(' ')
+    },
+    shift: {
+        valType: 'number',
+        editType: 'plot',
+        description: [
+            'Moves the axis a given number of pixels from where it would have been otherwise.',
+            'If `autoshift=true`, then this defaults to a padding of `-3` if `side=left`',
+            'and `+3` if `side=right`. Defaults to `0` if `autoshift=false`.',
```

Comment:
Negative padding could be very confusing . Instead we should internally handle this sign adjustment depending on the `side`.
Also IMHO the `shift` should have a `min` set to zero.
@alexcjohnson @hannahker

Reply:
If it were just a padding I'd agree with you @archmoj, but when `autoshift=false` it can be used to move the axis either direction - so it must be signed, therefore I think for consistency the padding variant should keep the signed behavior. We can add to the documentation though that negative values move it to the left and positive values move it to the right.

Reply:
@hannahker please add a commit to improve the documentation. Thank you!

Reply:
Done!